### PR TITLE
rootless: not require userns for help/version

### PR DESCRIPTION
**Pull request by giuseppe**
these commands do not require to be root in an userns

Closes: https://github.com/containers/libpod/issues/1263

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
`podman help` and `podman version` work but `podman info` gives me the following error:
```
could not get runtime: chown /home/vrothberg/.local/share/containers/storage/vfs: operation not permitted
```

**Comment by vrothberg** [comment]
@giuseppe Can we add integration tests for those three cases? Just to avoid any potential regressions.

File: cmd/podman/main.go [hunk 1]
```diff
@@ -25,19 +25,24 @@ var (
 	exitCode = 125
 )
 
+var cmdsNotRequiringRootless = []string{
+	"help",
+	"version",
+}
+
+func sliceContains(slice []string, s string) bool {
```
Comment (TomSweeneyRedHat) [inline cmd/podman/main.go]: This function probably should live in utils.go.

Comment (giuseppe) [reply cmd/podman/main.go]: moved

**Comment by vrothberg** [comment]
@giuseppe `podman info` isn't working with the latest commit.

File: cmd/podman/main.go [hunk 2]
```diff
@@ -25,19 +25,15 @@ var (
 	exitCode = 125
 )
 
+var cmdsNotRequiringRootless = []string{
+	"help",
```
Comment (rhatdan) [inline cmd/podman/main.go]: Why not make this a map, then you don't need to the string compare stuff below,

**Comment by giuseppe** [comment]
done, I've updated the PR title

**Comment by vrothberg** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**